src/database/DataTypes/Transaction.js

```diff
@@ -446,7 +446,7 @@ Transaction.schema = {
     serialNumber: { type: 'string', default: 'placeholderSerialNumber' },
     otherParty: { type: 'Name', optional: true },
     comment: { type: 'string', optional: true },
-    entryDate: { type: 'date', default: new Date() },
+    entryDate: { type: 'date', optional: true },
```

Comment:
I think the original issue was for `Transactions` but I guess extending this to `Requisition`'s is good also, as per your tests. At the moment, though you've only covered the `Transaction` case, so a `Requisition` incoming would still try to assign `null` to the required `entryDate` field which will throw an error.

Reply:
Yeah, I mentioned that in the PR. As I understood it, the idea was to hotfix patch the issue to prevent any buggy behaviour in the wild. I'm not sure if this is something we want to actually go with long-term, which is why I haven't extended it to any other cases/objects yet.

Reply:
As I understand it, this change leaves open the question of whether we want to integrate these changes further in future. From what I gathered, the idea was to hotfix patch the issue to prevent any buggy behaviour in the wild. I'm not sure if this is something we want to actually go with long-term, which is why I haven't extended it to any other cases/objects yet.

Reply:
sorry, I mis-read - I guess it just seemed like with the test cases and adding to requisitions code to handle `null` dates it seemed like it had just been forgotten. 

Without this change in the `Requisition` model, there can't be `null` entry dates. So I think either adding it to the `Requisition` model, or removing the code from the requisition pages seems correct? I would go for adding it to the `Requisition` model also, as `Transactions` shouldn't be coming in hot with `null` `entryDate` fields, but they are, so might happen with `Requisition` also. This does sound like a slippery slope though :)

Reply:
Currently the changes are only to `Transaction` stuff, no added code to `Requisitions` yet. It's quite easy to hotfix some additional changes to `Requisition` at a later date. Since we're not 100% on how mobile should handle malformed dates and haven't isolated the cause of the issue on the desktop side, it might be better to touch as little of the codebase as possible for now. OK with other suggestions, though.